app/src/main/java/com/cornellappdev/volume/ui/viewmodels/PublicationsViewModel.kt

```diff
@@ -37,20 +38,59 @@ class PublicationsViewModel @Inject constructor(
     fun followPublication(publicationSlug: String) = viewModelScope.launch {
         val uuid = userPreferencesRepository.fetchUuid()
         userRepository.followPublication(publicationSlug, uuid)
+        createUiState()
+    }
+    
+    fun unfollowPublication(publicationSlug: String)=viewModelScope.launch {
+        val uuid=userPreferencesRepository.fetchUuid()
+        userRepository.unfollowPublication(publicationSlug, uuid)
+        createUiState()
+    }
 
-        // move from more to followed
+    private fun createUiState()=viewModelScope.launch{
+        val allPublications= publicationRepository.fetchAllPublications()
+        val followedPublicationsSlugs =
+            userRepository.getUser(userPreferencesRepository.fetchUuid()).followedPublicationSlugs
+        val followedPublications = allPublications.filter {
+            followedPublicationsSlugs.contains(it.slug)
+        }
+        val morePublicationsState = allPublications.filter {
+            !followedPublicationsSlugs.contains(it.slug)
+        }
+        publicationsUiState = publicationsUiState.copy(
+            followedPublicationsState = PublicationsRetrievalState.Success(
+                followedPublications
+            ),
+            morePublicationsState = PublicationsRetrievalState.Success(
+                morePublicationsState
+            )
+        )
     }
 
     private fun queryAllPublications() = viewModelScope.launch {
         try {
+            val allPublications= publicationRepository.fetchAllPublications()
+            val followedPublicationsId =
+                userRepository.getUser(userPreferencesRepository.fetchUuid()).followedPublicationSlugs
+
+            val followedPublications = allPublications.filter {
+                followedPublicationsId.contains(it.slug)
+            }
+            val morePublicationsState = allPublications.filter {
+                !followedPublicationsId.contains(it.slug)
+            }
             publicationsUiState = publicationsUiState.copy(
-                publicationsState = PublicationsRetrievalState.Success(
-                    publicationRepository.fetchAllPublications()
+                followedPublicationsState = PublicationsRetrievalState.Success(
+                    followedPublications
+                ),
```

Comment:
These two do the same thing, one can be used over the other (probably queryAllPublications since it has the try-catch)